### DISCUSSION TITLE
Consider dropping xcov as a requirement and use more "nvim native" tools.

### DISCUSSION THREAD

**Discussion by andreacfromtheapp** [wojciech-kulik/xcodebuild.nvim#10] [Ideas]
Hi @wojciech-kulik 

first things first: I'm still learning Swift so I'm light years away from making any real use of your amazing work. That said, I'm a long time vi/vi/nvim and now AstroNvim user and would like `xcov` to be an optional install instead of a requirement. To leave space to more native tools. Besides the "native feel", what really bothers me is that `xcov` seems to fetch the entire gems universe with it. Including some `telnet` gems. No thanks. I don't really like to have such a huge dependency and so many gems. telnet? "_What could possibly go wrong?!_"

Instead, hear me out, since this is a nvim-centric project, I'd rather rely on:

- https://github.com/nvim-neotest/neotest (doesn't support Swift directly but..)
- https://github.com/nvim-neotest/neotest-vim-test (...this can bridge vim-test to the above...)
- https://github.com/vim-test/vim-test (..and vim-test supports Swift)
- https://github.com/andythigpen/nvim-coverage (this may lack the fancy browser reports, but it's an idea).

As a side note, I recently started useing AstroNvim and will implement your IDE(as) in it as language pack. I'm a regular contributor with a few PRs for integrations in a just a few days of use. One of them was XBase but ehm, your looks way better on paper.

If I knew Lua well enough I'd contribute, but ehm... one language at the time.

Cheers

**Comment by wojciech-kulik** [comment]
Hello,

thank you for the feedback!

> would like xcov to be an optional install instead of a requirement

Actually, it is optional if you don't use code coverage which is by default disabled in the config. This information was missing in the README, but I already updated it :)!

`xcov` was used temporarily, because I couldn't find any information on how to export `xccovarchive` required to get code coverage per line. Xcode tools are extremely badly designed and are missing essential documentation. I went through all the documentation and there is not a single information on how to do that. 

However, recently by just reading code on GitHub, I discovered how other tools export that (which also wasn't a trivial task because most tools rely on other tools, but finally I got to the root), and now I can implement it using native tools. I'm going to do that in near future. 

Then I will create my own report using just nvim buffer which should be even more comfortable to use because it wouldn't require switching between Neovim and a browser.

> Instead, hear me out, since this is a nvim-centric project, I'd rather rely on:
> 
> https://github.com/nvim-neotest/neotest (doesn't support Swift directly but..)
> https://github.com/nvim-neotest/neotest-vim-test (...this can bridge vim-test to the above...)
> https://github.com/vim-test/vim-test (..and vim-test supports Swift)

Unfortunately, as far as I can see `vim-test` only supports Swift Packages, not Xcode projects which are much more complex to process.

> https://github.com/nvim-neotest/neotest (doesn't support Swift directly but..)

This tool looks great. However, I think it might not be that easy to integrate it. As I said, Xcode tools are very bad and non-standardized that's why most plugins don't support it, and that's why my plugin exists. In this case, discovering test location is quite a challenging task because it requires for example knowledge about targets. File with the same name could exist in multiple locations and the name of the file doesn't have to match the name of the test class from logs ([more info here](https://github.com/wojciech-kulik/xcodebuild.nvim#-test-file-search---file-matching)).

However, this plugin also looks really cool, especially I like the test tree which is missing in my plugin right now. I will check out if the integration is possible without losing functionality.

> https://github.com/andythigpen/nvim-coverage (this may lack the fancy browser reports, but it's an idea).

`nvim-coverage` looks great, I will consider using it!

Thank you very much for the really great feedback. I will let you know here once I make any progress with those things :)!

**Reply by andreacfromtheapp** [reply]
Hi @wojciech-kulik 

thank you for your message, I'm relieved to know `xcov` is optional already!

Also, thank you for being receptive to my feedback, I appreciate it.

I will start a preliminary AstroNvim language pack with xcdeobuild then, to see how far I can go with this.

Looking forward to native coverage tools!

Cheers

**Reply by wojciech-kulik** [reply]
@gacallea resolved in #11. Now `xcov` is not needed anymore. I also implemented native code coverage report using `nui.nvim` to display it ;)

**Reply by andreacfromtheapp** [reply]
@wojciech-kulik WOW!! that was fast! thank you so much <3

**Comment by wojciech-kulik** [comment]
Resolved in #11
